internal/pkg/blob/gcs.go

```diff
@@ -86,3 +113,15 @@ func (b *BlobGCP) Delete(ctx context.Context, path string) error {
 func (b *BlobGCP) Close() error {
 	return b.bucket.Close()
 }
+
+// SignUrl returns an open accessible signed url for the given blob key
+func (b *BlobGCP) SignUrl(ctx context.Context, key string, ttlInSeconds int64, contentType string, method string) (string, error) {
+
+	strOpts := &storage.SignedURLOptions{
+		Method:      method,
+		Expires:     time.Now().Add(time.Duration(ttlInSeconds) * time.Second),
+		ContentType: contentType,
+	}
+
+	return b.bucketHandle.SignedURL(key, strOpts)
```

Comment:
Was there a reason to use the GCS library's bucket handle here instead of the bucket handle's [SignedURL](https://pkg.go.dev/gocloud.dev/blob#Bucket.SignedURL) already available in `gocloud.dev/blob`?

Reply:
yeah, GCS library containing bucket handle provide us a way to automatically import the credentials, the `SignedUrl` version from the `gocloud.dev/blob` doesn't.

here's the issue that I saw about it. https://github.com/googleapis/google-cloud-go/pull/4604